### PR TITLE
Misc gunmod fixes

### LINKED ISSUE #15743: Misleading integrated mod message
Trying to un-mod a gun with integrated mods fails with a message indicating that the gun isn't modded.

Some of the integrated mods can be applied to an existing gun, for example bore mods.

### PR DESCRIPTION
**Pull request by mugling**
A collection on minor fixes for gunmods:
- Removal of loaded gunmods now correctly handles unloading of ammo (via `game::unload`)
- Incidental fix in `game::unload` (91b623d)
- Order of messages is adjusted (so you unload first and then remove a mod)
- Correctly distinguishes between no mods and no removable mods (fixes #15743)
- Correctly resets mode upon removal of gunmods (most of #15375)
- Don't require unloading before adjusting mods (superfluous now `bore` mods are `IRREMOVABLE`)

### CODE REVIEW

File: src/player.cpp [hunk 1]
```diff
@@ -10696,28 +10697,16 @@ void player::use(int inventory_position)
...
-            add_msg( _( "You remove all the modifications from your %s." ), used->tname().c_str() );
-
-            for( int i = mods.size() - 1; i >= 0; --i ) {
-                if( mods[i].is_gunmod() && !mods[i].has_flag( "IRREMOVABLE" ) ) {
-                    remove_gunmod( used, i );
-                }
-            }
+        if( prompt.ret >= 0 ) {
+            item *gm = mods[ prompt.ret ];
+            std::string name = gm->tname().c_str();
```
Comment (BevapDin) [inline src/player.cpp]: The `.c_str()` is not needed.

Comment (mugling) [inline src/player.cpp]: > The .c_str() is not needed.

Fixed in 8973911

**Comment by Coolthulhu** [comment]
Minor: one of the `game::unload`s returns a void, the other bool.

**Comment by mugling** [comment]
`game::unload( item& )` should really become `item::unload( player &p )`

**Pull request merged**